src/app/taquin/taquin.component.ts

```diff
@@ -0,0 +1,67 @@
+import { Component, OnInit, OnChanges, Input } from '@angular/core';
+import { shuffle, isEqual } from 'lodash';
+
+@Component({
+  selector: 'app-taquin',
+  templateUrl: './taquin.component.html',
+  styleUrls: ['../app.component.css'],
+})
+export class TaquinComponent implements OnInit, OnChanges {
+  @Input() gridSize: number;
+  TilesOrder: Array<number>;
+  SizeStr: string;
+  hasWon = false;
+  counter = 0;
+  constructor() { }
+
+  ngOnInit(): void {
+    this.SizeStr = this.gridSize.toString();
+    this.TilesOrder = shuffle([...Array(this.gridSize).keys()]);
+  }
+
+  ngOnChanges(): void {
+    this.SizeStr = this.gridSize.toString();
+    this.TilesOrder = shuffle([...Array(this.gridSize).keys()]);
+  }
+
+  sourceContent(content: number): string {
+    const src: string = 'assets/' + this.SizeStr + '/';
+    if (content !== 0) {
+      return src + String(content) + '.jpg';
+    } else {
+      return src + 'final.jpg';
+    }
```

Comment:
No need for an else statement since you return a result
```suggestion
    if (content !== 0) {
      return `${src}${content}.jpg`;
    }
    return `${src}final.jpg`;
```